### ISSUE TITLE
[JIT] Can't use ndim in script

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#23890]
```
import torch

def foo(x):
    if x.ndim > 3:
        return torch.neg(x)
    else:
        return x


x = torch.ones(3, 4)
x2 = torch.ones(3, 4, 5, 6)

print(foo(x))
print(foo(x2))

scripted = torch.jit.script(foo)
```

```
import torch

def foo(x):
    if x.ndim > 3:
        return torch.neg(x)
    else:
        return x


x = torch.ones(3, 4)
x2 = torch.ones(3, 4, 5, 6)

print(foo(x))
print(foo(x2))

scripted = torch.jit.script(foo)

```

**Comment by eellison** [comment]
You can call `x.dim()` in script

**Comment by jamesr66a** [comment]
@eellison yeah but this should still work

**Comment by eellison** [comment]
`ndim` is deprecated & isn't in the [docs](https://pytorch.org/docs/stable/tensors.html). And the error message for this will show `dim` in the list of similar operators

**Comment by jamesr66a** [comment]
huh? it's on the docs:

![image](https://user-images.githubusercontent.com/4685384/62815869-bf307300-bad3-11e9-8460-331931e0dd25.png)

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
